crates/sui-graphql-rpc/src/types/event.rs

```diff
@@ -72,16 +76,40 @@ impl Event {
     }
 
     /// Addresses of the senders of the event
-    async fn senders(&self) -> Result<Option<Vec<Address>>> {
-        let mut addrs = Vec::with_capacity(self.stored.senders.len());
-        for sender in &self.stored.senders {
-            let Some(sender) = &sender else { continue };
+    async fn senders(
+        &self,
+        ctx: &Context<'_>,
+        first: Option<u64>,
+        after: Option<CSender>,
+        last: Option<u64>,
+        before: Option<CSender>,
+    ) -> Result<Connection<String, Address>> {
+        let page = Page::from_params(ctx.data_unchecked(), first, after, last, before)?;
+        let mut connection = Connection::new(false, false);
+        let Some((prev, next, cs)) = page.paginate_indices(self.stored.senders.len()) else {
+            return Ok(connection);
+        };
+
+        connection.has_previous_page = prev;
+        connection.has_next_page = next;
+
+        for c in cs {
+            let sender = &self.stored.senders[*c]
```

Comment:
(You could use the `Some(Some(...)))` trick here -- but probably not relevant in light of not needing a connection for this specific endpoint).